Cubical/HITs/SetTruncation/Properties.agda

```diff
@@ -92,3 +107,53 @@ setSigmaIso {A = A} {B = B} = iso fun funinv sect retr
   retr : retract fun funinv
   retr = elim (λ _ → isOfHLevelPath 2 setTruncIsSet _ _)
               λ { _ → refl }
+
+sigmaElim : ∀ {ℓ ℓ'} {B : ∥ A ∥₂ → Type ℓ} {C : Σ ∥ A ∥₂ B  → Type ℓ'}
+            (Bset : (x : Σ ∥ A ∥₂ B) → isSet (C x))
+            (g : (a : A) (b : B ∣ a ∣₂) → C (∣ a ∣₂ , b))
+            (x : Σ ∥ A ∥₂ B) → C x
+sigmaElim {B = B} {C = C} set g (x , y) = elim {B = λ x → (y : B x) → C (x , y)}
+                                               (λ _ → isOfHLevelΠ 2 λ _ → set (_ , _))
+                                               g x y
+
+sigmaProdElim : ∀ {ℓ ℓ' ℓ''} {B : Type ℓ} {C : ∥ A ∥₂ × ∥ B ∥₂ → Type ℓ'} {D : Σ (∥ A ∥₂ × ∥ B ∥₂) C  → Type ℓ''}
+             (Bset : (x : Σ (∥ A ∥₂ × ∥ B ∥₂) C) → isSet (D x))
+             (g : (a : A) (b : B) (c : C (∣ a ∣₂ , ∣ b ∣₂)) → D ((∣ a ∣₂ , ∣ b ∣₂) , c))
+             (x : Σ (∥ A ∥₂ × ∥ B ∥₂) C) → D x
+sigmaProdElim {B = B} {C = C} {D = D} set g ((x , y) , c) =
+  elim {B = λ x → (y : ∥ B ∥₂) (c : C (x , y)) → D ((x , y) , c)}
+       (λ _ → isOfHLevelΠ 2 λ _ → isOfHLevelΠ 2 λ _ → set _)
+       (λ x → elim (λ _ → isOfHLevelΠ 2 λ _ → set _)
+                    λ y c → g x y c)
+       x y c
+
+
+prodElim : ∀ {ℓ ℓ' ℓ''} {A : Type ℓ} {B : Type ℓ'} {C : ∥ A ∥₂ × ∥ B ∥₂ → Type ℓ''}
+        → ((x : ∥ A ∥₂ × ∥ B ∥₂) → isOfHLevel 2 (C x))
+        → ((a : A) (b : B) → C (∣ a ∣₂ , ∣ b ∣₂))
+        → (x : ∥ A ∥₂ × ∥ B ∥₂) → C x
+prodElim {A = A} {B = B} {C = C} hlevel ind (a , b) = schonf a b
+  where
+  schonf : (a : ∥ A ∥₂) (b : ∥ B ∥₂) → C (a , b)
+  schonf = elim (λ x → isOfHLevelΠ 2 λ y → hlevel (_ , _)) λ a → elim (λ x → hlevel (_ , _))
+                 λ b → ind a b
+
+prodElim2 : ∀ {ℓ ℓ' ℓ'' ℓ''' ℓ''''} {A : Type ℓ} {B : Type ℓ'} {C : Type ℓ''} {D : Type ℓ'''}
```

Comment:
Can you please move all of these levels to a private variable at the top and make the definitions in the file use them?